src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,328 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.fastq.FastqRecord;
+import htsjdk.samtools.fastq.FastqWriter;
+import htsjdk.samtools.fastq.FastqWriterFactory;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.StructuralVariationSparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+import scala.Tuple2;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+// TODO: log stderr messages from sga
+// TODO: choose which parameters allowed to be tunable
+// TODO: not all steps in SGA are currently covered
+// TODO: if throws, would temp files be cleaned up automatically?
+// TODO: how to log processes' progression in Spark? Conrrently only information when things go seriously wrong is logged in exception messages.
+@CommandLineProgramProperties(
+        summary        = "Program to call SGA for performing local assembly and outputs assembled contigs.",
+        oneLineSummary = "Perform SGA-based local assembly on fasta files on Spark",
+        programGroup   = StructuralVariationSparkProgramGroup.class)
+public final class RunSGANaivelySpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+
+    @Argument(doc       = "A path to a directory to write results to.",
+              shortName = "outDir",
+              fullName  = "outputDirectory",
+              optional  = false)
+    public String outputDir = null;
+
+    @Argument(doc       = "Number of threads to use when running sga.",
+              shortName = "t",
+              fullName  = "threads",
+              optional  = true)
+    public int threads = 1;
+
+    @Argument(doc       = "To run k-mer based read correction, filter and duplication removal in SGA or not.",
+              shortName = "correct",
+              fullName  = "correctNFilter",
+              optional  = true)
+    public boolean runCorrectionSteps = false;
+
+    @Override
+    public boolean requiresReads(){
+        return true;
+    }
+
+    @Override
+    public void runTool(final JavaSparkContext ctx){
+
+        final JavaRDD<GATKRead> reads = getReads();
+        // TODO: experimental interface to caller, should be getting this from JavaSparkContext
+        final JavaPairRDD<Long, Iterable<GATKRead>> readsAroundBreakpoints = reads.groupBy( read -> 1L);
```

Comment:
I know this is just here for testing purposes but you could also test this by setting up a pair RDD in another tool that looks the way you want it to and saving it with something like saveObjectFile(), and then loading that saved RDD in this tool. That would let you test running this tool on multiple breakpoints. Just a suggestion in case you're looking for a way to test that, don't feel like you need to do it unless you want to.